src/typeguard/_functions.py

```diff
@@ -61,21 +91,32 @@ def check_type(value: object, expected_type: Any) -> Any:
 
     :param value: value to be checked against ``expected_type``
     :param expected_type: a class or generic type instance
+    :param forward_ref_policy: see :attr:`TypeCheckConfiguration.forward_ref_policy`
+    :param typecheck_fail_callback:
+        see :attr`TypeCheckConfiguration.typecheck_fail_callback`
+    :param collection_check_strategy:
+        see :attr:`TypeCheckConfiguration.collection_check_strategy`
     :return: ``value``, unmodified
     :raises TypeCheckError: if there is a type mismatch
 
     """
+    config = TypeCheckConfiguration(
+        forward_ref_policy=forward_ref_policy,
+        typecheck_fail_callback=typecheck_fail_callback,
+        collection_check_strategy=collection_check_strategy,
+    )
```

Comment:
This overrides all of the configuration, even if the user just wants to override a single option.

Reply:
Not really, the defaults gathered from the defaults from of `TypeCheckConfiguration`: https://github.com/agronholm/typeguard/blob/2f4c104770adcc3b0d53bbc52a7f05bf787808b2/src/typeguard/_functions.py#L71

But the semantics are different from `@typechecked` and import hook, in that this is completely detached from the global config. My reasoning is that `check_type` is likely to be used differently from those, like in my case with phantom-types, and in this context it doesn't make sense to have shared mutable state. Does that make sense? :)

Reply:
> Not really, the defaults gathered from the defaults from of `TypeCheckConfiguration`:

The issue is that it disregards the current global config entirely, unlike how it works in `master`. This is one breaking change I'm not sure I want to merge.

Reply:
That's correct, as I mentioned in the comment above, I want to challenge the choice to make this function's behavior depend on shared mutable state. I understand why it makes sense for `@typechecked` and the import hook to have a global config object, but I think it's unwise for this function to depend on it.

Consider if these field are made to default to the global config, in order to fulfill the case where you don't want to be affected by the globally shared state, then you would always have to pass all configuration. What's worse, if typeguard adds more configuration options to this function in the future, it would be very hard for a user to detect that these have been added and that they should update all their calls to include a static default value.

I.e., my opinion is that `check_type` is sufficiently different from the instrumenting APIs to be treated differently.

Reply:
Alright, you're starting to convince me. This change would have to be reflected in the documentation too then. And `@typechecked` needs to have the overrides added to it too.